pkg/handler/collector/git/git.go

```diff
@@ -0,0 +1,179 @@
+//
+// Copyright 2022 The GUAC Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package git_collector
+
+import (
+	"context"
+	"os"
+	"time"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/guacsec/guac/pkg/handler/collector"
+	"github.com/guacsec/guac/pkg/handler/collector/file"
+	"github.com/guacsec/guac/pkg/handler/processor"
+	"github.com/guacsec/guac/pkg/logging"
+	"go.uber.org/zap"
+)
+
+const (
+	CollectorGit = "GIT"
+)
+
+type gitCol struct {
+	url           string
+	dir           string
+	lastChecked   time.Time
+	poll          bool
+	interval      time.Duration
+	fileCollector *file.FileCollector
+}
+
+func NewGitCol(ctx context.Context, url string, dir string, poll bool, interval time.Duration) *gitCol {
+	logger := logging.FromContext(ctx)
+	fileCollector := file.NewFileCollector(ctx, dir, false, time.Second)
+	err := collector.RegisterDocumentCollector(fileCollector, file.FileCollectorType)
+	if err != nil {
+		logger.Errorf("unable to register file collector: %v", err)
+	}
+
+	return &gitCol{
+		url:           url,
+		dir:           dir,
+		poll:          poll,
+		interval:      interval,
+		fileCollector: fileCollector,
+	}
+}
+
+// RetrieveArtifacts collects the documents from the collector. It emits each collected
+// document through the channel to be collected and processed by the upstream processor.
+// The function should block until all the artifacts are collected and return a nil error
+// or return an error from the collector crashing. This function can keep running and check
+// for new artifacts as they are being uploaded by polling on an interval or run once and
+// grab all the artifacts and end.
+func (g *gitCol) RetrieveArtifacts(ctx context.Context, docChannel chan<- *processor.Document) error {
```

Comment:
nit: Do we wanna call this gitCollector spelled out. Might make it easier to search in the code.